### PR TITLE
[CI][Bisect][1] Skeleton for automated bisect of release tests

### PR DESCRIPTION
**Pull request by can-anyscale**
## Why are these changes needed?
A script to bisect release test failures. This PR only contains a skeleton and unit-tests

## Related issue number

Issue #34243

## Checks

- [X] I've signed off every commit(by using the -s flag, i.e., `git commit -s`) in this PR.
- [X] I've run `scripts/format.sh` to lint the changes in this PR.
- [X] I've made sure the tests are passing. Note that there might be a few flaky tests, see the recent failures at https://flakey-tests.ray.io/
- Testing Strategy
   - [X] Unit tests

### CODE REVIEW

File: release/ray_release/scripts/ray_bisect.py [hunk 1]
```diff
@@ -0,0 +1,42 @@
...
+@click.argument("failing_commit", required=True, type=str)
+def main(test_name: str, passing_commit: str, failing_commit: str) -> None:
+    blamed_commit = _bisect(test_name, passing_commit, failing_commit)
+    print(f"Blamed commit found: {blamed_commit}")
+
+
+def _bisect(test_name: str, passing_commit: str, failing_commit: str) -> str:
+    lists = _get_commit_lists(passing_commit, failing_commit)
+    while len(lists) > 1:
+        middle_commit = lists[math.floor(len(lists) / 2)]
```
Comment (krfricke) [inline release/ray_release/scripts/ray_bisect.py:19]: nit
```suggestion
        middle_commit = lists[len(lists) // 2]
```

File: release/ray_release/scripts/ray_bisect.py [hunk 2]
```diff
@@ -0,0 +1,42 @@
...
+        is_passing = _run_test(test_name, middle_commit)
+        if is_passing:
+            passing_commit = middle_commit
+        else:
+            failing_commit = middle_commit
+        lists = _get_commit_lists(passing_commit, failing_commit)
```
Comment (krfricke) [inline release/ray_release/scripts/ray_bisect.py:25]: Can we just modify the existing commit list? I.e. something like (pseudo code)

```
middle_commit_idx = len(lists) // 2
middle_commit = lists[middle_commit_idx]
# ...
if is_passing:
    lists = lists[middle_commit_idx+1:]
else:
    lists = lists[:middle_commit_idx]
```

Comment (krfricke) [reply release/ray_release/scripts/ray_bisect.py:25]: This would also simplify the unit test below...

**Review by krfricke — COMMENTED** [review]
Some questions

**Review by krfricke — APPROVED** [review]
This LGTM! Linter seems to be failing

**Comment by ollie-iterators** [comment]
The documentation test should be fixed by merging in recent changes.

**Pull request merged**